core-bundle/src/DependencyInjection/ContaoCoreExtension.php

```diff
@@ -116,6 +117,21 @@ private function handleSearchIndexer(array $config, ContainerBuilder $container)
             $defaultIndexer = $container->getDefinition('contao.search.indexer.default');
             $defaultIndexer->setArgument(2, $config['search']['indexProtected']);
         }
+
+        // Configure search index listener
+        switch ($config['search']['listener']) {
```

Comment:
What if `$config['search']['default_indexer']['enable']` is `false`?

Reply:
That‘s not a possible configuration value🤷‍♂️

Reply:
We are not preventing this combination, so ATM it is possible. Should we throw an exception in this case? Or automatically remove the `contao.listener.search_index` definition?

Reply:
What makes you think we do not prevent this? It can only be one of the predefinded 4 strings. It‘s an enum.

Reply:
```yml
contao:
    search:
        default_indexer:
            enable: false
        listener: index_and_delete
```

Reply:
How is the listener related to the indexer? They share zero logic and responsibilities. This PR is about the listener. You cannot set `false` for the listener.

Reply:
The listener has an **index**_and_delete option and it calls `$this->indexer->index($document);`. How can the listener not be related to the indexer?

Reply:
Yeah but they do not relate at all? The default indexer can be disabled and yet still you want to have the listener call `index()` or `delete()` on your custom implementation.
We could, however, disable the listener if there was no indexer registered at all. I don't know where to do that though. It would need to be in the `SearchIndexerPass`. But isn't it strange to remove a `kernel.response` listener in the compiler pass? Probably not...hmmm :D

Reply:
And also, it's not related to this PR.

Reply:
Implemented in https://github.com/contao/contao/pull/983/commits/792a6badf498db3a4f869e104fdca70244997388.